src/melee.cpp

```diff
@@ -179,8 +210,39 @@ bool player::handle_melee_wear( item &shield, float wear_multiplier )
         g->m.add_item_or_charges( pos(), elem );
     }
 
+    // Preserve item temporarily for component breakdown
+    item temp = shield;
+
     remove_item( shield );
 
+    // Breakdown fragile weapons into components
+    if( temp.has_flag( "FRAGILE_MELEE" ) && !temp.components.empty() ) {
+        add_msg_player_or_npc( m_bad, _( "Your %s breaks apart!" ),
+                                      _( "<npcname>'s %s breaks apart!" ),
+                                      str.c_str() );
+
+        std::vector<item> all_comps = temp.components;
+
+        for( auto &comp : all_comps ) {
+            int break_chance = comp.typeId() == weak_comp ? 2 : 8;
```

Comment:
I'm not sure what you're doing here.
`weak_comp` is the item with the highest `chip_resistance`. Why does this specific item get 4 times lower chance at avoiding destruction?

Reply:
You have that backwards.  `weak_comp` has the lowest chip resistance, and so is considered the "breaking point" of the weapon and has a lower chance of being recovered when the weapon falls apart.